core/kernel/notif.c

```diff
@@ -127,6 +411,14 @@ void notif_register_driver(struct notif_driver *ndrv)
 	SLIST_INSERT_HEAD(&notif_driver_head, ndrv, link);
 
 	cpu_spin_unlock_xrestore(&notif_lock, old_itr_status);
+
+#if defined(CFG_CORE_IT_NOTIF)
+	old_itr_status = cpu_spin_lock_xsave(&it_lock);
+
+	bit_nset(notif_it_masked, 0, (int)NOTIF_IT_VALUE_MAX);
```

Comment:
Why is this done each time we register a new driver to be notified of an event?

Reply:
This is the main async notif registering function (`notif_register_driver()`), not the itr notif registering function (`notif_it_register()`). The later masks its own itr with `set_it_mask(id, 1);`.